File: Python/8py 1/count_by.py
"""
|--------------------------------------------------------------------------
| Count by X
|--------------------------------------------------------------------------
|
"""

def count_by(x, n):
   return [x * i for i in range(1, n+1)]


def count_by(x, n):
   return range(x, x * n + 1, x)


def count_by(x, n):
   return [e for e in range(x, x * n + x, x)]


print(count_by(1, 5), [1, 2, 3, 4, 5])
print(count_by(2, 5), [2, 4, 6, 8, 10])
print(count_by(3, 5), [3, 6, 9, 12, 15])
print(count_by(50, 5), [50, 100, 150, 200, 250])

File: PHP/8php 1/countBy.php
<?php
/*
|--------------------------------------------------------------------------
| Count by X
|--------------------------------------------------------------------------
|
*/

function countBy($x, $n) {
   $res = [];
   for ($i=1; $i <= $n; $i++) { 
      array_push($res, $x * $i);
   }
   return $res;
}


function countBy($x, $n) {
   return range($x, $n * $x, $x);
}


function countBy($x, $n) {
   $res = []; $i = 1;
   while ($i <= $n){
      $res[] = $x * $i;
      $i++;
   }
   return $res;
}


function countBy(int $x, int $n): array {
   return array_map(
      function(int $num) use (&$x): int {
         return $num * $x;
      },
      range(1, $n)
   );
}


print_r(countBy(1, 10));
print_r(countBy(2, 5));